core/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/typing/TypeExtractor.java

```diff
@@ -73,8 +73,13 @@ else if (SubmodelElementList.class.isAssignableFrom(type)) {
             if (Objects.nonNull(submodelElementList.getTypeValueListElement())
                     && (submodelElementList.getTypeValueListElement() == AasSubmodelElements.SUBMODEL_ELEMENT_COLLECTION
                             || submodelElementList.getTypeValueListElement() == AasSubmodelElements.SUBMODEL_ELEMENT_LIST)) {
-                for (int i = 0; i < submodelElementList.getValue().size(); i++) {
-                    builder.element(Integer.toString(i), extractTypeInfoForSubmodelElement(submodelElementList.getValue().get(i)));
+                if (submodelElementList.getValue().size() == 1) {
```

Comment:
What is the reasoning for this? Why should we not store the index if a list only contains a single element?

Reply:
This was not done by me, you changed that